src/RestWrite.js

```diff
@@ -341,6 +341,11 @@ RestWrite.prototype.transformUser = function() {
     });
   }
 
+  // If we're updating a _User object, clear the user cache for the session
```

Comment:
I mean when we do the login shortcut with oAuth, we go through that method. If there is a cached token for the found user, we should destroy it too to remove any wrong possible state, the same way we remove it upon user login